support/BRFileService.c

```diff
@@ -127,21 +212,96 @@ fileServiceCreate (const char *basePath,
     if (NULL == dir) return NULL;
     closedir(dir);
 
-    // Create the directory hierarchy
-    size_t pathToTypeSize = strlen(basePath) + 1 + strlen(currency) + 1 + strlen(network) + 1;
-    char dirPath[pathToTypeSize];
+    // Create the file service itself
+    BRFileService fs = calloc (1, sizeof (struct BRFileServiceRecord));
+    fs->sdb = NULL;
+    fs->sdbPath = NULL;
+
+    // Save currency and network
+    fs->currency = strdup (currency);
+    fs->network  = strdup (network);
+
+    // Locate the SQLITE Database
+    size_t sdbPathLength = strlen (basePath) + 1 + strlen (FILE_SERVICE_SDB_FILENAME) + 1;
+    fs->sdbPath = malloc (sdbPathLength);
+    sprintf (fs->sdbPath, "%s/%s", basePath, FILE_SERVICE_SDB_FILENAME);
+
+    // Create/Open the SQLITE Database
+    sqlite3_status_code status = sqlite3_open(fs->sdbPath, &fs->sdb);
+    if (SQLITE_OK != status) {
+        if (NULL != fs->sdb) sqlite3_close(fs->sdb);
+        if (NULL != fs->sdbPath) free (fs->sdbPath);
+        free (fs);
+
+        // Set error info
+        return NULL;
+    }
 
-    sprintf (dirPath, "%s/%s", basePath, currency);
-    if (-1 == directoryMake(dirPath)) return NULL;
+    // Create the SQLite 'Entity' Table
+    sqlite3_stmt *sdbCreateTableStmt;
+    status = sqlite3_prepare_v2 (fs->sdb, FILE_SERVICE_SDB_ENTITY_TABLE, -1, &sdbCreateTableStmt, NULL);
+    if (SQLITE_OK != status) {
+        // ...
+        printf ("foo\n");
+//        return NULL;
+    }
+    if (SQLITE_DONE != sqlite3_step(sdbCreateTableStmt)) {
+        printf ("bar\n");
+ //       return NULL;
+    }
+    sqlite3_finalize(sdbCreateTableStmt);
 
-    sprintf(dirPath, "%s/%s/%s", basePath, currency, network);
-    if (-1 == directoryMake(dirPath)) return NULL;
+    FileServiceSQL sql;
 
-    BRFileService fs = calloc (1, sizeof (struct BRFileServiceRecord));
+    // Create the SQLITE 'Insert into Entity' Statement
```

Comment:
We are leaking, in all the return NULL cases.

Reply:
Yes, pending error processing